### PR TITLE
Update and expand comparison doc.

### PR DESCRIPTION
**Pull request by brian-brazil**
Add Nagios and Sensu, update existing ones.

### CODE REVIEW

File: content/docs/introduction/comparison.md [hunk 1]
```diff
@@ -63,76 +63,111 @@ samples at arbitrary intervals as scrapes or rule evaluations occur. Since new
...
+
+  * If you're doing event logging
+  * Commercial option offers clustering for InfluxDB, which is also better for long term data storage
+  * Eventually consistent view of data between replicas
+
+Where Prometheus is better:
+
+  * If you're primarily doing metrics
+  * More powerful query language, alerting, and notification functionality
+  * Higher availability and uptime for graphing and alerting
```
Comment (beorn7) [inline content/docs/introduction/comparison.md]: I would end each bullet point and above with a period.

File: content/docs/introduction/comparison.md [hunk 2]
```diff
@@ -160,6 +197,95 @@ simple aggregations via its API.
...
+
+
+## Prometheus vs. Nagios
+
+[Nagios](https://www.nagios.org/) is a monitoring system that originated in the
+90s as NetSaint.
+
+### Scope
+
+Nagios is primarily about alerting based on the exit codes of scripts. These are called "checks".
```
Comment (beorn7) [inline content/docs/introduction/comparison.md]: Typography nit: Use “...” to quote.

File: content/docs/introduction/comparison.md [hunk 3]
```diff
@@ -160,6 +197,95 @@ simple aggregations via its API.
...
+[Sensu](https://sensuapp.org/) is broadly speaking a more modern Nagios.
+
+### Scope
+
+The same general scope differences as in the case of
+[Nagios](/docs/introduction/comparison/#prometheus-vs-nagios) apply here.
+
+The primary difference is that Sensu clients [register
+themselves](https://sensuapp.org/docs/0.27/reference/clients.html#what-is-a-sensu-client),
+and can determine the checks to run either from local or local configuration.
```
Comment (beorn7) [inline content/docs/introduction/comparison.md]: local or local?

File: content/docs/introduction/comparison.md [hunk 4]
```diff
@@ -160,6 +197,95 @@ simple aggregations via its API.
...
+### Storage
+
+Sensu has storage in Redis called stashes, this is used primarily for storing
+silences. It also stores all the clients that have registered with it.
+
+### Architecture
+
+Sensu has a [number of
+components](https://sensuapp.org/docs/0.27/overview/architecture.html). It uses
+RabbitMQ as a transport, Redis for current state and a separate Server for
```
Comment (beorn7) [inline content/docs/introduction/comparison.md]: Comma after "state".

File: content/docs/introduction/comparison.md [hunk 5]
```diff
@@ -160,6 +197,95 @@ simple aggregations via its API.
...
+Both RabbitMQ and Redis can be clustered. Multiple copies of the server can be
+run for scaling and redundancy.
+
+### Summary
+
+If you have an existing Nagios setup that you wish to scale as-is or taking
+advantage of the registration feature of Sensu, then Sensu is a good choice.
+
+If you want to do whitebox monitoring, or have a very dynamic or cloud based
+environment then Prometheus is a good choice.
```
Comment (beorn7) [inline content/docs/introduction/comparison.md]: Comma after "environment".

File: content/docs/introduction/comparison.md [hunk 6]
```diff
@@ -63,76 +63,111 @@ samples at arbitrary intervals as scrapes or rule evaluations occur. Since new
 samples are simply appended, old data may be kept arbitrarily long. Prometheus
 also works well for many short-lived, frequently changing sets of time series.
 
+### Summary
+
+Prometheus offers a richer data model and query language, in addition to be
```
Comment (juliusv) [inline content/docs/introduction/comparison.md]: be -> being

File: content/docs/introduction/comparison.md [hunk 7]
```diff
@@ -148,10 +183,12 @@ The same scope differences as in the case of
 
 OpenTSDB's data model is almost identical to Prometheus's: time series are
 identified by a set of arbitrary key-value pairs (OpenTSDB "tags" are
-Prometheus "labels"). There are minor differences though, such as that
-Prometheus allows arbitrary characters in label values, while OpenTSDB is more
-restrictive. OpenTSDB is also lacking a full query language, only allowing
-simple aggregations via its API.
+Prometheus "labels"). All data for a metric is [stored
+together](http://opentsdb.net/docs/build/html/user_guide/writing/index.html#time-series-cardinality),
+limiting cardinally of metrics. There are minor differences though,
```
Comment (juliusv) [inline content/docs/introduction/comparison.md]: cardinally -> the cardinality

File: content/docs/introduction/comparison.md [hunk 8]
```diff
@@ -160,6 +197,95 @@ simple aggregations via its API.
...
+There are plugins which can store data such as [for visualisation](https://docs.pnp4nagios.org/).
+
+### Architecture
+
+Nagios servers are standalone. All configuration of checks is via file.
+
+### Summary
+
+Nagios is suitable for basic monitoring of small and/or static systems where
+blackbox probing is sufficient.
```
Comment (juliusv) [inline content/docs/introduction/comparison.md]: "... and no historical metrics data is needed."

File: content/docs/introduction/comparison.md [hunk 9]
```diff
@@ -160,6 +197,95 @@ simple aggregations via its API.
...
+
+There is also a [client socket](https://sensuapp.org/docs/0.27/reference/clients.html#what-is-the-sensu-client-socket) permitting arbitrary check results to be pushed into Sensu.
+
+### Data model
+
+Sensu has the same rough data model as [Nagios](/docs/introduction/comparison/#prometheus-vs-nagios).
+
+### Storage
+
+Sensu has storage in Redis called stashes, this is used primarily for storing
```
Comment (juliusv) [inline content/docs/introduction/comparison.md]: Run-on sentence: "stashes. This is used"

File: content/docs/introduction/comparison.md (hunk 4 above)
Comment (juliusv) [inline content/docs/introduction/comparison.md]: There's no Oxford commas throughout. I'd be a big fan of adding them everywhere though :P

**Comment by juliusv** [comment]
Looks good besides comments.

File: content/docs/introduction/comparison.md (hunk 1 above)
Comment (brian-brazil) [inline content/docs/introduction/comparison.md]: done

File: content/docs/introduction/comparison.md (hunk 2 above)
Comment (brian-brazil) [inline content/docs/introduction/comparison.md]: Ah yes, I keep on forgetting about the smart quote key on my keyboard :)

File: content/docs/introduction/comparison.md (hunk 3 above)
Comment (brian-brazil) [inline content/docs/introduction/comparison.md]: fixed

File: content/docs/introduction/comparison.md (hunk 4 above)
Comment (brian-brazil) [inline content/docs/introduction/comparison.md]: Added comma

File: content/docs/introduction/comparison.md (hunk 5 above)
Comment (brian-brazil) [inline content/docs/introduction/comparison.md]: done

File: content/docs/introduction/comparison.md (hunk 7 above)
Comment (brian-brazil) [inline content/docs/introduction/comparison.md]: done

File: content/docs/introduction/comparison.md (hunk 8 above)
Comment (brian-brazil) [inline content/docs/introduction/comparison.md]: Not sure about this. It doesn't have it inherently, but there's plenty of common ways it is done which I've referred to above.

File: content/docs/introduction/comparison.md (hunk 9 above)
Comment (brian-brazil) [inline content/docs/introduction/comparison.md]: Fixed, and changed to plural.

**Comment by brian-brazil** [comment]
Thanks for the comments, PR is updated.

**Comment by beorn7** [comment]
I'll leave final check to Julius's keen eyes.

**Comment by juliusv** [comment]
:+1:

**Pull request merged**